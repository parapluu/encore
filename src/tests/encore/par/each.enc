def presetArray(size: int): [int]
  let arr = new [int](size)
  in {
    for i in [0..size-1] arr[i] = i;
    arr;
  }


class Main
  check_arr: [bool]
  size: int

  def setup(): void {
    this.size = 2000;
    this.check_arr = new [bool](this.size);
    for v in [0..this.size] this.check_arr[v] = false;
  }

  def test_setting_array(): void
    let arr = presetArray(this.size) in
    each(arr) >> (\(i: int) -> { this.check_arr[i] = true; i; } )
              >> \(i: int) -> assertTrue(this.check_arr[i])

  def test_split_words_and_print(): void {
    let words = (new String("this is a test")).split(" ")
    in (each(words)) >> (\(w: String) -> print("Word: {}\n", w))
  }

  def test_array_of_one(): void{
    let arr = presetArray(1) in
      each(arr) >> (\(x: int) -> print("result: {}", x + 1))
  }

  def main(): void {
    this.setup();
    this.test_setting_array();
    this.test_split_words_and_print();
    this.test_array_of_one();
  }
