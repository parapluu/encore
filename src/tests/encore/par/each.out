Word: this
Word: is
Word: a
Word: test
result: 1
